cmd/olm/main.go

```diff
@@ -82,6 +85,33 @@ func init() {
 	feature.AddFlag(pflag.CommandLine)
 }
 
+type validatingRoundTripper struct {
+	delegate http.RoundTripper
+}
+
+func (rt *validatingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	if req.Method == "POST" {
+		b, err := req.GetBody()
+		if err != nil {
+			panic(err)
```

Comment:
Why `panic` vs `return nil, err`? will this kill the process?

Reply:
I wanted the stacktrace to clearly show where the offending code was. Yes, this does cause process exit. If we want to keep this around, I'd make it gated on some dev-only flag (`$CI=true`).

Reply:
I'd say yes, keep it around, and gate on CI

Reply:
Done